cpu/sam0_common/periph/spi.c

```diff
@@ -184,3 +184,25 @@ void spi_transfer_bytes(spi_t bus, spi_cs_t cs, bool cont,
         gpio_set((gpio_t)cs);
     }
 }
+
+void spi_pm_cb_enter(int deep)
+{
+    if (deep) {
+        for (size_t i = 0; i < SPI_NUMOF; i++) {
+            spi_t bus = SPI_DEV(i);
+            gpio_disable_mux(spi_config[bus].mosi_pin);
+            gpio_disable_mux(spi_config[bus].clk_pin);
+        }
+    }
+}
+
+void spi_pm_cb_leave(int deep)
+{
+    if (deep) {
+        for (size_t i = 0; i < SPI_NUMOF; i++) {
+            spi_t bus = SPI_DEV(i);
+            gpio_init_mux(spi_config[bus].mosi_pin, spi_config[bus].mosi_mux);
+            gpio_init_mux(spi_config[bus].clk_pin, spi_config[bus].clk_mux);
```

Comment:
The problem here is that not all SPI buses from the board configuration are necessarily used by the application, so those pins might be used for something else.

I don't see any register that we can reliably query here, so how about introducing a `static uint8_t spi_enabled;` bit field that gets set in `spi_init()`

    spi_enabled |= 1 << bus;

Then you can skip those entries that are not enabled in the loop here and in `spi_pm_cb_enter`.

Reply:
Good idea. This bitmask may become useful, once `periph_spi_reconfigure` has been merged and `spi_deinit_pins()` gets called.